From 6f3b92773bf3354191ec352abc55c5ab456842d0 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 21 Jan 2022 04:34:03 +0200
Subject: [PATCH] [SW-69190] arch: update general region macro

Currently there are 2 macros used to define base of general region
HL_SDRAM_BASE (Gen region base in arc viewpoint), HL_SRAM_BASE (Gen
region base in 64 bit).

In the existing platforms, the address space for memories in general
region were not strictly in GEN_BASE + offset format.

Updating the macros to address it as general region base instead of
SRAM, SDRAM for better readability. The offsets are adjusted to
the correct GEN_BASE to follow the GENBASE + offset format.

Change-Id: I38b43165e365229f934ea1988f4af39ee3deee6f
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index e0dd757..af41271 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -260,7 +260,7 @@
 
 	recovery_img_size = spi_flash_readl(0);
 	dma_copy_image(recovery_img_size, 0,
-		HL_SRAM_BASE + HL_ICCM_OFFSET, SPI_DMA_DST_HBW);
+		HL_GEN_MEM_BASE_FULL + HL_ICCM_OFFSET, SPI_DMA_DST_HBW);
 	dma_wait_finish();
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_RECOVERY_DMA);
 
@@ -401,7 +401,7 @@
 	arcboot_ram_size = (uint16_t)spi_flash_readl(flash_arcboot_ram_offset);
 
 	dma_copy_image(arcboot_ram_size, flash_arcboot_ram_offset,
-		       HL_SRAM_BASE + HL_ICCM_OFFSET, SPI_DMA_DST_HBW);
+			HL_GEN_MEM_BASE_FULL + HL_ICCM_OFFSET, SPI_DMA_DST_HBW);
 	if (dma_wait_finish() == DMA_CRC_ERROR)
 		return ARCBOOT_BRINGUP_ERROR;
 	arcboot_rom_status_update_bringup(image_progress, PROGRESS_BRINGUP_DMA);
diff --git a/arch/gaudi2/arc/soc.h b/arch/gaudi2/arc/soc.h
index 04a8713..4b5eec5 100644
--- a/arch/gaudi2/arc/soc.h
+++ b/arch/gaudi2/arc/soc.h
@@ -22,25 +22,21 @@
 #define FLASH_VALID_OFFSET		0
 #define FLASH_VALID_MAX_SIZE		0x200000
 
-#define HL_SRAM_BASE			0x1000007FF0000000ULL
+#define HL_GEN_MEM_BASE_FULL		0x1000007FF4000000ULL
 #define HL_SP_SRAM_BASE			0x1000007FF7FE0000ULL
 #define HL_SRAM_PHY_BASE		0x1000FFFFFD000000ULL
 
-/*
- * This is not SDRAM Base, but the base address of General region
- * into which the rest of the components are mapped
- */
-#define HL_SDRAM_BASE			0x20000000
+#define HL_GEN_MEM_BASE			0x24000000
 #define HL_SRAM_LOG_BASE		0x11000000
 
-#define HL_FLASH_OFFSET			0x6000000
-#define HL_ICCM_OFFSET			0x7040000
-#define HL_SBROM_OFFSET			0x7FC0000
-#define HL_FWSRAM_OFFSET		0x7FD0000
-#define HL_SCRATCHPAD_SRAM_OFFSET	0x7FE0000
-#define HL_PCIE_FWSRAM_OFFSET		0x7FF0000
+#define HL_FLASH_OFFSET			0x2000000
+#define HL_ICCM_OFFSET			0x3040000
+#define HL_SBROM_OFFSET			0x3FC0000
+#define HL_FWSRAM_OFFSET		0x3FD0000
+#define HL_SCRATCHPAD_SRAM_OFFSET	0x3FE0000
+#define HL_PCIE_FWSRAM_OFFSET		0x3FF0000
 #define HL_CFG_BASE			0xF8000000
-#define HL_ICCM_BASE			(HL_SRAM_BASE + HL_ICCM_OFFSET)
+#define HL_ICCM_BASE			(HL_GEN_MEM_BASE_FULL + HL_ICCM_OFFSET)
 
 #define HL_ICCM_SIZE			0x20000
 #define HL_SP_SRAM_SIZE			0x10000
diff --git a/arch/gaudi3/arc/soc.h b/arch/gaudi3/arc/soc.h
index 02619c9..5c8c17e 100644
--- a/arch/gaudi3/arc/soc.h
+++ b/arch/gaudi3/arc/soc.h
@@ -69,11 +69,7 @@
 #define HL_FW_MEM_BASE			REGION_BASE_ADDR(REGION_FW_MEM_IDX)
 #define HL_FW_MEM_SIZE			SZ_256MB
 
-/*
- * This is not SDRAM Base, but the base address of General region
- * into which the rest of the components are mapped
- */
-#define HL_SDRAM_BASE			HL_FW_MEM_BASE
+#define HL_GEN_MEM_BASE			HL_FW_MEM_BASE
 
 #define HL_FLASH_OFFSET			0x2000000
 #define HL_ARC0_ICCM_OFFSET		0x3000000
@@ -89,9 +85,8 @@
 #define HL_PCIE_FWSRAM_OFFSET		0x7F00000
 
 #define HL_FW_MEM_BASE_FULL		0x0300007FD0000000ull
-/* TODO: Fix the macro name for HL_SRAM_BASE in common code.*/
-#define HL_SRAM_BASE			HL_FW_MEM_BASE_FULL
-#define HL_SP_SRAM_BASE			(HL_FW_MEM_BASE_FULL + HL_SCRATCHPAD_SRAM_OFFSET)
+#define HL_GEN_MEM_BASE_FULL		HL_FW_MEM_BASE_FULL
+#define HL_SP_SRAM_BASE			(HL_GEN_MEM_BASE_FULL + HL_SCRATCHPAD_SRAM_OFFSET)
 #define HL_SP_SRAM_SIZE			0x10000
 #define HL_MIRROR_CFG_BASE		HL_D2D_LBW_BASE
 
diff --git a/arch/greco/arc_elf32/soc.h b/arch/greco/arc_elf32/soc.h
index 8ef900d..bf96f88 100644
--- a/arch/greco/arc_elf32/soc.h
+++ b/arch/greco/arc_elf32/soc.h
@@ -26,15 +26,16 @@
 #define FLASH_VALID_OFFSET			0
 #define FLASH_VALID_MAX_SIZE			0x1000000
 
-#define HL_SRAM_BASE				0x7FF0000000ULL
+#define HL_GEN_MEM_BASE				0x28000000
+#define HL_GEN_MEM_BASE_FULL			0x7FF8000000ULL
 #define HL_SP_SRAM_BASE            		0x7FFBFE0000ULL
-#define HL_SDRAM_BASE				0x10000000
-#define HL_FLASH_OFFSET				0xA000000
-#define HL_ICCM_OFFSET				0xB020000
-#define HL_SBROM_OFFSET				0xB040000
-#define HL_FWSRAM_OFFSET			0xB050000
-#define HL_SCRATCHPAD_SRAM_OFFSET		0xBFE0000
-#define HL_PCIE_FWSRAM_OFFSET			0xBFF0000
+#define HL_GEN_MEM_BASE				0x28000000
+#define HL_FLASH_OFFSET				0x2000000
+#define HL_ICCM_OFFSET				0x3020000
+#define HL_SBROM_OFFSET				0x3040000
+#define HL_FWSRAM_OFFSET			0x3050000
+#define HL_SCRATCHPAD_SRAM_OFFSET		0x3FE0000
+#define HL_PCIE_FWSRAM_OFFSET			0x3FF0000
 
 #define HL_CFG_BASE				0xFC000000
 
diff --git a/include/arcboot.h b/include/arcboot.h
index 4be5777..a2e18e8 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -25,13 +25,13 @@
 #define BIT(nr)					(1 << (nr))
 #define ARRAY_SIZE(x)				(sizeof(x) / sizeof((x)[0]))
 
-#define HL_FLASH_BASE				(HL_SDRAM_BASE + \
+#define HL_FLASH_BASE				(HL_GEN_MEM_BASE + \
 						 HL_FLASH_OFFSET)
-#define HL_SBROM_BASE				(HL_SDRAM_BASE + \
+#define HL_SBROM_BASE				(HL_GEN_MEM_BASE + \
 						 HL_SBROM_OFFSET)
-#define HL_FWSRAM_BASE				(HL_SDRAM_BASE + \
+#define HL_FWSRAM_BASE				(HL_GEN_MEM_BASE + \
 						 HL_FWSRAM_OFFSET)
-#define HL_SCRATCHPAD_SRAM_BASE			(HL_SDRAM_BASE + \
+#define HL_SCRATCHPAD_SRAM_BASE			(HL_GEN_MEM_BASE + \
 						 HL_SCRATCHPAD_SRAM_OFFSET)
 #define ARCBOOT_PRIMARY				0
 #define ARCBOOT_SECONDARY			1
